aten/src/ATen/native/CPUFallback.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <ATen/core/ivalue.h>
+#include <ATen/core/stack.h>
+#include <ATen/core/boxing/KernelFunction.h>
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <torch/library.h>
+
+namespace at { namespace native {
+
+// This function implements a boxed fallback to CPU.
+// External backends can add their own custom logging on top if it to customize their own CPU fallbacks.
+TORCH_API void cpu_fallback(const c10::OperatorHandle& op, torch::jit::Stack* stack);
+
+template<c10::KernelFunction::BoxedKernelFunction* fallback_fn, class ReturnType, class... ParameterTypes>
+TORCH_API ReturnType call_fallback_fn(const char* name, const char* overload_name, ParameterTypes... args) {
+    auto op = c10::Dispatcher::singleton()
+        .findSchemaOrThrow(name, overload_name)
+        .typed<ReturnType (ParameterTypes...)>();
+    return c10::impl::BoxedKernelWrapper<ReturnType(ParameterTypes...)>::call(
+        c10::KernelFunction::make_boxed_function<fallback_fn>,
+        nullptr,
+        op,
+        c10::DispatchKeySet(), // we know that the cpu_fallback doesn't use the dispatch keyset.
+        std::forward<ParameterTypes>(args)...
+    );
+}
+
+// Version for ops with no overload name.
+template<c10::KernelFunction::BoxedKernelFunction* fallback_fn, class ReturnType, class... ParameterTypes>
+TORCH_API ReturnType call_fallback_fn(const char* name, ParameterTypes... args) {
+    return call_fallback_fn<fallback_fn, ReturnType, ParameterTypes...>(name, "", args...);
+}
+
+template<c10::KernelFunction::BoxedKernelFunction* fallback_fn, class F2>
+struct call_fallback_fn2 final {};
```

Comment:
Similarly, I'd add a comment here describing why you'd use this one instead of `call_fallback_fn`

Reply:
I should have preemptively commented these. Once https://github.com/pytorch/pytorch/pull/58092 lands I can kill the  version above, and use this nicer version (and rename `call_fallback_fn2` to `call_fallback_fn`). I talked with Ed and there's probably an even nicer version of this where you just call `at::native::call_fallback_fn<&xla_cpu_fallback, at::_ops::_adaptive_avg_pool3d>::call(self, output_size);`, and `at::_ops::_adaptive_avg_pool3d` has enough information to give you both the op schema type and the op's name + overload name, that way users of this API only have to pass in the arguments + the fallback that they want to call.

Right now I'm planning on just landing the PR independently from that change and doing a fix-up later.